### ISSUE TITLE
Property keyboardShouldPersistTaps={true} not working in <Modal /> view

### ISSUE THREAD

**Issue by ogtfaber** [facebook/react-native#10138]
### Issue Description

Given a <TextInput /> in a <Modal />, and the TextInput being active (keyboard is shown), I can't get any Touchable... element to respond to onPress before dismissing the keyboard.
### Steps to Reproduce / Code Snippets

See the example code below. I would expect the TouchableOpacity (a yellow square) to popup an Alert while the TextInput is active and the keyboard is shown. Instead, I have to tap to dismiss the keyboard first. 

The property keyboardShouldPersistTaps={true} does the trick in another view, but not inside of a <Modal />

```
<Modal
  animationType={"slide"}
  transparent={false}
  visible={this.props.visible} >
  <ScrollView
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps={true} >
    <TouchableOpacity onPress={() => alert(1)} style={{ backgroundColor: 'yellow', width: 200, height: 200, }} />
    <TextInput style={{
      height: 40,
      backgroundColor: 'red',
    }} />
  </ScrollView>
</Modal>
```
### Additional Information
- React Native version: 0.34
- Platform(s): iOS
- Operating System: macOS

**Comment by ilyaivjev** [comment]
+1,  0.35.0-rc.0

**Comment by manggit** [comment]
Same problem here, using a list view
0.36 iOS

**Comment by hlynn93** [comment]
+1 0.36 on iOS

**Comment by Arkanine** [comment]
Have the same issue.

**Comment by roysG** [comment]
Same issue : react native - 0.37

**Comment by dcolin** [comment]
Same problem 0.39, any idea?

**Comment by chapati23** [comment]
confirmed, same issue. `keyboardShouldPersisTaps` has no effect on modals.
has anyone found a workaround for this by any chance?

**Comment by razvan-tudosa** [comment]
I lost 2 hours because of this bug.

**Comment by kenma9123** [comment]
I can confirm this is not working :/ using RN 0.41.1

**Comment by dxiao** [comment]
putting the scrollview on the outside of my Modal did not solve issue for me.

**Comment by despreston** [comment]
+ 1

**Comment by andrzli** [comment]
Has anyone found a solution for this?

**Comment by wv1124** [comment]
here: https://github.com/facebook/react-native/issues/8180

solution:
`android:windowSoftInputMode="adjustPan"`

**Comment by iiitmahesh** [comment]
:+1: Same issues

**Comment by despreston** [comment]
what fixed it for me is to make sure that all nested scrollviews have this property set.

e.g:

`<ScrollView><ScrollView keyboardShouldPersistTaps></ScrollView></ScrollView>`

This won't work because the parent ScrollView does not have this property set.

**Comment by gre** [comment]
the fix for me was that I had to set `keyboardShouldPersistTaps` on ALL ancestors of my Modal...
otherwise, if any ScrollView in your parent tree do the default keyboardShouldPersistTaps="none", the keyboard will be dismissed (in my case I want to prevent a dismiss).

anyway, whatever what you want to solve (dismiss/not dismiss/respond to onPress) –  you will have to be away that the `<Modal>` cares about its parent stack, I assumed it would be, like rendered at top level, but no. rendering your modal component from root OR within a complex stack of scrollview, have different behavior.

**Comment by Guacaswole** [comment]
@despreston & @gre that solved it for me.

Once my both my parent and child `FlatList` had the keyboardShouldPersistTaps prop set it worked a treat

**Comment by kakakalot** [comment]
I just use `TouchableHighlight` instead of `TouchableOpacity` and the `onPress` works

```
          <ScrollView
            keyboardShouldPersistTaps='always'
            style={styles.listItem}>
            {items.map((item, index) => this._renderTouchableItem(item, index))}
          </ScrollView>

  _renderTouchableItem = (item, index) => {
    return (
      <TouchableHighlight
        key={index}
        underlayColor='transparent'
        onPress={() => this.props.onTouchableItemSelected(this, index)}
      >
        <Text value={item.label} style={{ padding: 16, paddingLeft: 24 }} />
      </TouchableHighlight>
    )
  }
```

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.

**Comment by profiVideos** [comment]
I know this is closed.  However, I just struggled with this for a few hours as well.  Thanks to the comments read here; I found a solution that worked in my situation.  I didn't even know that Flatlist would accept the keyboardShouldPersistTaps='always' parameter.  I ended up having no extra ScrollViews, simply adding that parameter to Flatlist worked out great.

`      <FlatList
        keyboardShouldPersistTaps='always'
        data={this.props.itemList}
        extraData={this.props}
        renderItem={this.renderCardItem}
        ItemSeparatorComponent={this.itemSeparator}
      />
`
Cheers
